controllers/sr_controller/sr_controller.py

```diff
@@ -18,19 +18,83 @@
     "1470": 3,
 }
 
+STRICT_ZONES = {
+    "dev": (1, 2, 3),
+    "comp": (0, 1, 2, 3),
+}
+
 
 def get_robot_zone() -> int:
     return ROBOT_IDS_TO_CORNERS[os.environ['WEBOTS_ROBOT_ID']]
 
 
-def get_robot_file() -> Path:
-    zone_id = get_robot_zone()
-    robot_file_dir = ROOT.parent / ("zone-" + str(zone_id))
-    if robot_file_dir.is_dir():
-        return robot_file_dir / "robot.py"
-    if get_robot_mode() == "comp":
-        print("WARNING: Using default robot file location in competition mode.")
-    return ROOT.parent / "robot.py"
+def get_robot_file(zone_id: int, mode: str) -> Path:
+    """
+    Get the path to the proper robot.py file for zone_id and mode, ensuring that
+    it exists or exiting with a suitable error message.
+
+    The logic here is that:
+     - it is always an error for both a robot.py in the root and a zone-0 /
+       robot.py file to exist
+     - in competition mode: we check only for zone-X / robot.py files and error
+       if they are missing. We assume that this controller is not run at all for
+       zones which should not run.
+     - in development mode:
+        - zones 1-3 check only check only for zone-X / robot.py files and report
+          if they are missing but exit cleanly
+        - zone 0 checks for zone-0 / robot.py then a root robot.py. If neither
+          are found it copies an example into place (at the root) and uses that.
+    """
+
+    robot_file = ROOT.parent / "zone-{}".format(zone_id) / "robot.py"
+    fallback_robot_file = ROOT.parent / "robot.py"
+    strict_zones = STRICT_ZONES[mode]
+
+    if (
+        robot_file.exists() and
+        zone_id == 0 and
+        fallback_robot_file.exists()
+    ):
+        exit(
+            "Found robot controller in shared location and zone-0 location. "
+            "Remove one of the controllers before running the simulation\n"
+            "{}\n{}".format(robot_file, fallback_robot_file),
+        )
+
+    if zone_id in strict_zones:
+        if robot_file.exists():
+            return robot_file
+
+        print("No robot controller found for zone {}".format(zone_id))
+
+        # Only in competition mode is it an error for a robot file to be missing.
+        if mode == "comp":
+            exit(1)
+        else:
+            exit()
```

Comment:
This could be inlined and made a bit cleaner
```suggestion
        exit(1 if mode == "comp" else 0)
```

Reply:
I'm not convinced that this is clearer -- you've now got one line which is doing two things.

Reply:
Perhaps this might be better?
```suggestion
        missing_file_is_error = mode == "comp"
        exit(1 if missing_file_is_error else 0)
```
Though of course it could be simplified to just:
```suggestion
        missing_file_is_error = mode == "comp"
        exit(missing_file_is_error else)
```
by relying on `True` being `1` and `False` being `0`, though that might be a bit much.